app/assets/javascripts/angular-libs/angular.of.gardens.service.js

```diff
@@ -306,4 +307,103 @@ openFarmModule.directive('addToGardens', ['$rootScope', 'gardenService',
       },
       templateUrl: '/assets/templates/_add_to_gardens.html'
     }
-  }])
+  }]);
+
+openFarmApp.directive('addCrop', ['$http', 
+                                  '$rootScope', 
+                                  'cropService', 
+                                  'gardenService',
+  function addCrop($http, $rootScope, cropService, gardenService) {
+    return {
+      restrict: 'A',
+      scope: {
+        cropOnSelect: '=',
+        gardenQuery: '=',
+        query: '=',
+        user: '=',
+        objectType: '=',
+      },
+      controller: ['$scope', '$element', '$attrs',
+        function ($scope, $element, $attrs) {
+          $scope.placeholder = $attrs.placeholder || 'Search crops';
+          $scope.buttonValue = $attrs.buttonValue || 'Submit';
+          $scope.firstCrop = undefined;
+          $scope.finalCrop = undefined;
+          $scope.crops = undefined;
+          $scope.garden = undefined;
+          
+          //Typeahead search for crops
+          $scope.getCrops = function (val) {
+            // be nice and only hit the server if
+            // length >= 3
+            return $http.get('/api/v1/crops', {
+              params: {
+                filter: val
+              }
+            }).then(function(res) {
+              var crops = [];
+              crops = res.data.data;
+              if (crops.length === 0 && $scope.allowNew) {
+                crops.push({ attributes: {
+                  name: val,
+                  is_new: true
+                } });
+              }
+              crops = crops.map(function(crop) {
+                return cropService.utilities.buildCrop(crop, res.data.included);
+              });
+              $scope.firstCrop = crops[0];
+              $scope.crops = crops;
+              return crops;
+            })
+          };
+
+          //Typeahead search for crops
+          //cropSearch.getCrops("tomato");
+          $scope.addCropToGarden = function () {
+            $scope.crops1 = $scope.getCrops($scope.cropQuery);
+            var cropi;
+            if($scope.crops1) {
+            for (cropi in $scope.crops) {
```

Comment:
Have a look at this answer to see why this is necessary: http://stackoverflow.com/a/4166584/154392

Luckily we're using Angular, so you can use Angular's forEach to avoid this.

```
$scope.crops.forEach(function (value, key) {
  // use
});
```